app/model/PaymentSpecificData.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package model
+import play.api.libs.json._
+
+trait PaymentSpecificData {
```

Comment:
If I read correctly you intentions you wanted to use 
`sealed trait ....`, so all implementations will be known.

Reply:
We can add that , it will add as side benefit .  The main intention was being able to store more than one type of data